### PR TITLE
Allow low-level reader to read all data desc ids into a list of tables

### PR DESCRIPTION
**Pull request by keflavich**
@miguelcarcamov noted that one sometimes wants to read all DDIDs from a table, and looping through each DDID is quite slow because of the overhead in assembling the table metadata.

This allows the low-level reader to specify `all_ddids=True` and just loop over all the tables.

Example:
```python
from casa_formats_io.casa_low_level_io.table import CASATable
rslt = CASATable.read('HD163296_continuum.ms')
tables = rslt.as_astropy_table(all_ddids=True)
```

This is much faster than
```python
desc = Table.read('HD163296_continuum.ms/DATA_DESCRIPTION')
tables = [Table.read('HD163296_continuum.ms', data_desc_id=ii) for ii in desc['SPECTRAL_WINDOW_ID']]
```

### CODE REVIEW

**Comment by keflavich** [comment]
...well, I think it's faster? my tests are going slower now that I'm running `timeit`.

**Comment by keflavich** [comment]
```
%timeit tables = rslt.as_astropy_table(all_ddids=True)
30.4 s ± 222 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

File: casa_formats_io/casa_low_level_io/table.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def read(cls, filename, endian='<'):
 
         return table
 
-    def as_astropy_table(self, data_desc_id=None):
+    def as_astropy_table(self, data_desc_id=None, all_ddids=False):
```
Comment (astrofrog) [inline casa_formats_io/casa_low_level_io/table.py:310]: Maybe just make it so you can set ``data_desc_id='all'``?

**Review by astrofrog — CHANGES_REQUESTED** [review]
No objections from me but an API comment below

File: casa_formats_io/casa_low_level_io/table.py (hunk 1 above)
Comment (keflavich) [reply casa_formats_io/casa_low_level_io/table.py]: done

File: casa_formats_io/casa_low_level_io/table.py [hunk 2]
```diff
@@ -352,42 +352,20 @@ def as_astropy_table(self, data_desc_id=None):
 
             data_desc_ids = sorted(np.unique(table_columns['DATA_DESC_ID']))
 
-            tables = []
-
             if data_desc_id is None:
                 if len(data_desc_ids) == 1:
                     data_desc_id = data_desc_ids[0]
+                    return self._read_data_descid(table_columns, data_desc_ids)
```
Comment (keflavich) [inline casa_formats_io/casa_low_level_io/table.py]: this was supposed to be `data_desc_id`, not `data_desc_ids`

**Comment by astrofrog** [comment]
Can you rebase to resolve the conflict?

**Comment by keflavich** [comment]
I think I resolved them, but the rebase had several sets of conflicts that didn't make sense to me.  It looked like my commits were getting rebased in the wrong order.  I really have a hard time tracking rebases sometimes.

**Comment by astrofrog** [comment]
Not sure what happened with the rebase but it looks good now. I pushed a commit to fix the failing test which was unrelated to the changes here, not quite sure why it didn't occur before.

**Review by astrofrog — APPROVED** [review]

**Pull request merged**